Extensions/NotificationService/NotificationService.swift

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UserNotifications
+
+class NotificationService: UNNotificationServiceExtension {
+
+    var contentHandler: ((UNNotificationContent) -> Void)?
+    var bestAttemptContent: UNMutableNotificationContent?
+
+    override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
+        self.contentHandler = contentHandler
+        bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent)
+
+        var userInfo = request.content.userInfo
+        userInfo["url"] = "https://mozilla.org"
+        NSLog("NotificationService APNS NOTIFICATION \(userInfo)")
```

Comment:
NSLog appears in the Console.app. 

We can't see out logs when logging from the extension. (we can, but we want to be able to see both app and extension at the same time).

This is temporary code. It is currently tightly controlled so as not to be accessible from release.